### PR TITLE
feat(agui): propagate userId from forwardedProps to RuntimeContext

### PR DESCRIPTION
**Pull request by jujn**
## AgentScope-Java Version

[The version of AgentScope-Java you are working on, e.g. 1.0.12, check your pom.xml dependency version or run `mvn dependency:tree | grep agentscope-parent:pom`(only mac/linux)]

## Description

[Please describe the background, purpose, changes made, and how to test this PR]

## Checklist

Please check the following items before code is ready to be reviewed.

- [ ]  Code has been formatted with `mvn spotless:apply`
- [ ]  All tests are passing (`mvn test`)
- [ ]  Javadoc comments are complete and follow project conventions
- [ ]  Related documentation has been updated (e.g. links, examples, etc.)
- [ ]  Code is ready for review

### CODE REVIEW

File: agentscope-extensions/agentscope-extensions-protocol/agentscope-extensions-agui/src/main/java/io/agentscope/core/agui/adapter/AguiAgentAdapter.java [hunk 1]
```diff
@@ -173,6 +175,18 @@ private RuntimeContext buildRuntimeContext(RunAgentInput input) {
                 .build();
     }
 
+    private static String normalizeUserId(Object value) {
+        if (value == null) {
+            return null;
+        }
+        String userId = value.toString();
+        if (userId == null) {
+            return null;
```
Comment (AgentScopeJavaBot) [inline agentscope-extensions/agentscope-extensions-protocol/agentscope-extensions-agui/src/main/java/io/agentscope/core/agui/adapter/AguiAgentAdapter.java:184]: [nit] Dead code: `Object.toString()` is guaranteed by the Java Language Specification to never return `null`, so the `if (userId == null)` branch is unreachable. You can safely remove lines 183-185 and simplify to:
```java
private static String normalizeUserId(Object value) {
    if (value == null) {
        return null;
    }
    String userId = value.toString().trim();
    return userId.isEmpty() ? null : userId;
}
```

File: agentscope-extensions/agentscope-extensions-protocol/agentscope-extensions-agui/src/test/java/io/agentscope/core/agui/adapter/AguiAgentAdapterTest.java [hunk 2]
```diff
@@ -70,6 +70,71 @@ void setUp() {
...
+                        .forwardedProps(Map.of(AguiAgentAdapter.FORWARDED_PROP_USER_ID_KEY, 12345L))
+                        .build();
+
+        adapter.run(input).collectList().block();
+
+        RuntimeContext context = contextCaptor.getValue();
+        assertEquals("12345", context.getUserId());
+    }
+
+    @Test
```
Comment (AgentScopeJavaBot) [inline agentscope-extensions/agentscope-extensions-protocol/agentscope-extensions-agui/src/test/java/io/agentscope/core/agui/adapter/AguiAgentAdapterTest.java:117]: [nit] The `normalizeUserId` method has explicit trim + empty-check logic, but no test exercises the blank/whitespace-only userId edge cases (e.g. `""` or `"  "`). Consider adding a test like:
```java
@Test
void testRunNormalizesBlankUserIdToNull() {
    // ... setup ...
    RunAgentInput input = RunAgentInput.builder()
        .threadId("t").runId("r")
        .messages(List.of(AguiMessage.userMessage("m", "hi")))
        .forwardedProps(Map.of(AguiAgentAdapter.FORWARDED_PROP_USER_ID_KEY, "  "))
        .build();
    adapter.run(input).collectList().block();
    assertNull(contextCaptor.getValue().getUserId());
}
```
This would give full branch coverage for the normalization helper.

**Review by AgentScopeJavaBot — APPROVED** [review]
🤖 AI Review

This is a clean, well-scoped change that propagates the `userId` field from AG-UI protocol's `forwardedProps` into `RuntimeContext`, enabling downstream agent code to identify the calling user. The implementation follows existing patterns in `buildRuntimeContext`, includes a defensive `normalizeUserId` helper that handles null/non-string/blank values, and ships with three focused unit tests covering the happy path, non-string type coercion, and missing-key scenarios. The code is correct and ready to merge with two minor nits below.

File: agentscope-extensions/agentscope-extensions-protocol/agentscope-extensions-agui/src/main/java/io/agentscope/core/agui/adapter/AguiAgentAdapter.java (hunk 1 above)
Comment (AgentScopeJavaBot) [inline agentscope-extensions/agentscope-extensions-protocol/agentscope-extensions-agui/src/main/java/io/agentscope/core/agui/adapter/AguiAgentAdapter.java:184]: [nit] Dead code: `Object.toString()` is guaranteed by the Java Language Specification to never return `null`, so the `if (userId == null)` branch is unreachable. You can safely remove lines 183-185 and simplify to:
```java
private static String normalizeUserId(Object value) {
    if (value == null) {
        return null;
    }
    String userId = value.toString().trim();
    return userId.isEmpty() ? null : userId;
}
```

File: agentscope-extensions/agentscope-extensions-protocol/agentscope-extensions-agui/src/test/java/io/agentscope/core/agui/adapter/AguiAgentAdapterTest.java (hunk 2 above)
Comment (AgentScopeJavaBot) [inline agentscope-extensions/agentscope-extensions-protocol/agentscope-extensions-agui/src/test/java/io/agentscope/core/agui/adapter/AguiAgentAdapterTest.java:117]: [nit] The `normalizeUserId` method has explicit trim + empty-check logic, but no test exercises the blank/whitespace-only userId edge cases (e.g. `""` or `"  "`). Consider adding a test like:
```java
@Test
void testRunNormalizesBlankUserIdToNull() {
    // ... setup ...
    RunAgentInput input = RunAgentInput.builder()
        .threadId("t").runId("r")
        .messages(List.of(AguiMessage.userMessage("m", "hi")))
        .forwardedProps(Map.of(AguiAgentAdapter.FORWARDED_PROP_USER_ID_KEY, "  "))
        .build();
    adapter.run(input).collectList().block();
    assertNull(contextCaptor.getValue().getUserId());
}
```
This would give full branch coverage for the normalization helper.

**Review by AgentScopeJavaBot — APPROVED** [review]
🤖 AI Review

This is a clean, well-scoped change that propagates the `userId` field from AG-UI protocol's `forwardedProps` into `RuntimeContext`, enabling downstream agent code to identify the calling user. The implementation follows existing patterns in `buildRuntimeContext`, includes a defensive `normalizeUserId` helper that handles null/non-string/blank values, and ships with three focused unit tests covering the happy path, non-string type coercion, and missing-key scenarios. The code is correct and ready to merge with two minor nits below.

**Comment by zouyx** [comment]
@copilot+gpt-5.6-terra review

File: agentscope-extensions/agentscope-extensions-protocol/agentscope-extensions-agui/src/main/java/io/agentscope/core/agui/adapter/AguiAgentAdapter.java [hunk 3]
```diff
@@ -162,6 +163,7 @@ public Flux<AguiEvent> run(RunAgentInput input) {
     private RuntimeContext buildRuntimeContext(RunAgentInput input) {
         return RuntimeContext.builder()
                 .sessionId(input.getThreadId())
+                .userId(normalizeUserId(input.getForwardedProp(FORWARDED_PROP_USER_ID_KEY)))
```
Comment (zouyx) [inline agentscope-extensions/agentscope-extensions-protocol/agentscope-extensions-agui/src/main/java/io/agentscope/core/agui/adapter/AguiAgentAdapter.java:166]: 这样直接获取是否会存在身份伪造的风险？

**Review by oss-maintainer — APPROVED** [review]
### Summary
This PR contains 115 lines of changes. Review in progress.

---
*Automated review by github-manager-bot*

File: agentscope-extensions/agentscope-extensions-protocol/agentscope-extensions-agui/src/main/java/io/agentscope/core/agui/adapter/AguiAgentAdapter.java (hunk 3 above)
Comment (ningmeng0503) [reply agentscope-extensions/agentscope-extensions-protocol/agentscope-extensions-agui/src/main/java/io/agentscope/core/agui/adapter/AguiAgentAdapter.java:166]: 您说得很有道理，我完全认同RunAgentInput的参数确实需要校验，因为它们直接来自用户。不过我在想，是否可以把校验逻辑放在adapter之外的其他层来处理，这样职责会更清晰一些。

另外关于userId，我也觉得放在后台接口层从header或其他上下文中获取，再手动设置到RunAgentInput中，可能比让前端直接传递更合适，也更安全。

Comment (jujn) [reply agentscope-extensions/agentscope-extensions-protocol/agentscope-extensions-agui/src/main/java/io/agentscope/core/agui/adapter/AguiAgentAdapter.java:166]: @ningmeng0503 
Can you take charge of this? You can wait until the event mechanism is upgraded to 2.0. I originally wanted to add a submission to your PR, but it can't be edited. You can continue in: https://github.com/agentscope-ai/agentscope-java/pull/2044

Comment (ningmeng0503) [reply agentscope-extensions/agentscope-extensions-protocol/agentscope-extensions-agui/src/main/java/io/agentscope/core/agui/adapter/AguiAgentAdapter.java:166]: @jujn 
I'd be happy to. I'll continue to follow up in #2044.
